crates/pallet-domains/src/lib.rs

```diff
@@ -256,14 +286,58 @@ mod pallet {
 
     /// The domain registry
     #[pallet::storage]
-    pub(super) type DomainRegistry<T: Config> = StorageMap<
+    pub(super) type DomainRegistry<T: Config> =
+        StorageMap<_, Identity, DomainId, DomainObject<T::BlockNumber, T::AccountId>, OptionQuery>;
+
+    /// The domain block tree, map (`domain_id`, `domain_block_number`) to the hash of a domain blocks,
+    /// which can be used get the domain block in `DomainBlocks`
+    #[pallet::storage]
+    pub(super) type BlockTree<T: Config> = StorageDoubleMap<
         _,
         Identity,
         DomainId,
-        DomainObject<T::BlockNumber, T::Hash, T::AccountId>,
+        Identity,
+        T::DomainNumber,
+        BTreeSet<H256>,
+        ValueQuery,
+    >;
+
+    /// Mapping of domain block hash to domain block
+    #[pallet::storage]
+    pub(super) type DomainBlocks<T: Config> = StorageMap<
+        _,
+        Identity,
+        H256,
+        DomainBlock<T::BlockNumber, T::Hash, T::DomainNumber, T::DomainHash, BalanceOf<T>>,
         OptionQuery,
     >;
 
+    /// The head receipt number of each domain
+    #[pallet::storage]
+    pub(super) type HeadReceiptNumber<T: Config> =
+        StorageMap<_, Identity, DomainId, T::DomainNumber, ValueQuery>;
+
+    /// A set of `bundle_extrinsics_root` from all bundles that successfully submitted to the consensus
+    /// block, these extrinsics will be used to construct the domain block and `ExecutionInbox` is used
+    /// to ensure subsequent ERs of that domain block include all pre-validated extrinsic bundles.
+    #[pallet::storage]
+    pub(super) type ExecutionInbox<T: Config> =
+        StorageDoubleMap<_, Identity, DomainId, Identity, T::DomainNumber, Vec<H256>, ValueQuery>;
+
+    /// The block number of the best domain block, increase by one when the first bundle of the domain is
+    /// successfully submitted to current consensus block, which mean a new domain block with this block
+    /// number will be produce. Used as a pointer in `ExecutionInbox` to identify the current under building
+    /// domain block, also used as a mapping of consensus block number to domain block number.
+    #[pallet::storage]
+    pub(super) type HeadDomainNumber<T: Config> =
```

Comment:
I'm not able to find specific need for this storage item and `HeadReceiptNumber`. Can you help me understand how these are useful ?